charts/redpanda/helpers.go

```diff
@@ -92,7 +92,26 @@ func StatefulSetPodLabelsSelector(dot *helmette.Dot, statefulSet map[string]any)
 		"app.kubernetes.io/name":      Name(dot),
 	}
 
-	return helmette.Merge(commonLabels, defaults)
+	result := helmette.Merge(AdditionalSelectorLabels, defaults)
```

Comment:
nit: could this variable be called `selectors` or something a bit more descriptive? The `IsMapSubset` line is a lot more difficult to understand as I had to keep referring to the definition of `result`.